### ISSUE TITLE
Errors thrown from validate() prevent help text from being displayed

### ISSUE THREAD

**Issue by sharplet** [apple/swift-argument-parser#149]
My command accepts a list of paths as arguments, using `@Argument() var paths: [String]`. I couldn't find a built in way to validate that the array is non-empty, so I implemented `validate()` and throw a `ValidationError` in that case. Because CommandParser [validates commands before checking if help was requested](https://github.com/apple/swift-argument-parser/blob/223d62adc52d51669ae2ee19bdb8b7d9fd6fcd9c/Sources/ArgumentParser/Parsing/CommandParser.swift#L204), only a short usage string is ever displayed, instead of the full help text. 

**ArgumentParser version:** 0.0.6

**Swift version:**
```
Apple Swift version 5.2.2 (swiftlang-1103.0.32.6 clang-1103.0.32.51)
Target: x86_64-apple-darwin19.4.0
```

### Checklist
- [x] If possible, I've reproduced the issue using the `master` branch of this package
- [x] I've searched for [existing GitHub issues](https://github.com/apple/swift-argument-parser/issues)

### Steps to Reproduce

Paste this into a new Swift package:

```swift
import ArgumentParser

struct Test: ParsableCommand {
  @Argument(help: "A list of paths.")
  var paths: [String]

  func validate() throws {
    if paths.isEmpty {
      throw ValidationError("At least one path must be specified.")
    }
  }

  func run() throws {
    for path in paths {
      print(path)
    }
  }
}

Test.main()
```

Then run `swift run test --help`.

### Expected behavior

Expected the command to output the help text.

### Actual behavior

Received this output instead:

```
Error: At least one path must be specified.
Usage: test [<paths>  . . .]
```

**Comment by john-mueller** [comment]
I ran into a similar situation, where specifying both an `-h` flag and incorrect options prints an error instead of help. I would expect a help request to override any bad options/validation, which would take care of both issues. What do you think, @natecook1000?

**Comment by natecook1000** [comment]
@john-mueller I can go either way on `command -h --invalid-flag` — seems like printing a usage error is a reasonable response to that. `command -h` by itself really shouldn't ever fail.

**Comment by natecook1000** [comment]
Fixed by #166.
